File: src/main/java/com/ultreon/ultranlang/symbol/VarSymbol.java
package com.ultreon.ultranlang.symbol;

import com.ultreon.ultranlang.Repr;
import org.jetbrains.annotations.Nullable;

public class VarSymbol extends Symbol implements Repr {
    public VarSymbol(String name, @Nullable Symbol type) {
        super(name, type);
    }
}

File: src/main/kotlin/com/ultreon/ultranlang/symbol/VarSymbol.kt
package com.ultreon.ultranlang.symbol

import com.ultreon.ultranlang.ast.AST
import kotlin.reflect.KClass

class VarSymbol(name: String, type: Symbol?) : Symbol(name, type) {
    override fun toString(): String {
        return "<${this::class.qualifiedName}($name=$name, type=$type)>"
    }

    fun representation(): String = toString()
}